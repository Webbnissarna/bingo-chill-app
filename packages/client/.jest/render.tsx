import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
