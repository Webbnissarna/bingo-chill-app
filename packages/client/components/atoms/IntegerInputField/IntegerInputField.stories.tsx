import { useState } from "react";
import IntegerInputField from ".";
import type { StoryDefault } from "@ladle/react";

export function Default() {
  const [value, setValue] = useState(1337);
  return (
    <div className="bg-polarNight-1 p-5">
      <IntegerInputField value={value} onChange={setValue} />
    </div>
  );
}

export default {
  title: "Atoms / Integer Input Field",
} satisfies StoryDefault;
