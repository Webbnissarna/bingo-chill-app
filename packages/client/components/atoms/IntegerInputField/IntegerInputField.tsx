import { runCallbackIfInteger } from "@/utils/inputValidators";

interface IntegerInputFieldProps {
  value: number;
  onChange: (newValue: number) => void;
}

export default function IntegerInputField({
  value,
  onChange,
}: IntegerInputFieldProps): JSX.Element {
  return (
    <input
      className="w-full font-text text-base text-snowStorm-2 outline-none border-none shadow-none bg-polarNight-0 rounded px-2 py-1 focus:bg-polarNight-2 transition"
      type="number"
      value={value}
      onChange={(e) => runCallbackIfInteger(e.currentTarget.value, onChange)}
    />
  );
}
