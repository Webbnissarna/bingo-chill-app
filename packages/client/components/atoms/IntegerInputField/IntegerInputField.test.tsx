import { render } from "../../../.jest/render";
import IntegerInputField from ".";

describe("IntegerInputField", () => {
  it("renders", () => {
    const { asFragment } = render(
      <IntegerInputField value={0} onChange={() => undefined} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
