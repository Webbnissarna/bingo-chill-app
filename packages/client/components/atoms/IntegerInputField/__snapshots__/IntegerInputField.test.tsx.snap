// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntegerInputField renders 1`] = `
<DocumentFragment>
  <input
    class="w-full font-text text-base text-snowStorm-2 outline-none border-none shadow-none bg-polarNight-0 rounded px-2 py-1 focus:bg-polarNight-2 transition"
    type="number"
    value="0"
  />
</DocumentFragment>
`;
